@xen-orchestra/xapi/vm.js

```diff
@@ -129,9 +144,21 @@ class Vm {
     }
   }
 
-  async checkpoint(vmRef, { cancelToken = CancelToken.none, name_label } = {}) {
+  async checkpoint($defer, vmRef, { cancelToken = CancelToken.none, ignoreNobakVdis = false, name_label } = {}) {
+    const vm = await this.getRecord('VM', vmRef)
+
+    // cannot unplug VBDs on Running, Paused and Suspended VMs
+    if (ignoreNobakVdis && vm.power_state === 'Halted') {
+      await asyncMap(await listNobakVbds(this, vm.VBDs), async vbd => {
+        await this.VBD_destroy(vbd.$ref)
+        $defer.call(this, 'VBD_create', vbd)
+      })
+
+      ignoreNobakVdis = false
```

Comment:
I'm not convinced, the code will now have to check both booleans in order to decide what to do.

Reply:
I don't think it's a problem , since one of the boolean is the function parameter and the other means it's already handled.
Also, I think modifying parameters does not improve readability  https://eslint.org/docs/rules/no-param-reassign